tldraw/tldraw
# Improve image asset loading

This PR improves the loading cycle for image assets.
![Kapture 2024-06-29 at 17 30 54](https://github.com/tldraw/tldraw/assets/23072548/e4ec6355-8f25-44fc-bb60-930fb4072cbe)

Showing the correctly debounced upscale after resize ends
https://github.com/tldraw/tldraw/assets/23072548/58ca3682-daa4-4bea-8c0a-fb28a482a657

Previously, creating a new image for any reason (copying and pasting, alt-dragging) would have a 500ms delay before showing the image, even if the image had already been loaded. I thought this might be a bug with the way we preloaded images, but it turned out to be part of the logic added in #3684 for debouncing new images.

In this PR, we take better advantage of the signals library (and other state in tldraw) to allow immediate fetching when the image first comes up, and then correctly debounced fetching when the user finishes resizing a shape or moving the camera.

We no longer update the shape if the resulting URL is the same as it was before.

### Design decisions

- Don't adjust the LOD while resizing, even if the user has "paused" during a resize interaction

### Change type


### Test plan

1. Create an image shape
2. Duplicate it
3. The new image should immediately appear
4. Resize a shape
5. The LOD should change once you've finished resizing
4. Change the zoom level
5. The LOD should change once you've finished zooming

### Release notes

- Fixed a bug with new image shapes not immediately appearing.
